spec/helper/functional-test.helper.js

```diff
@@ -543,17 +407,26 @@ class FunctionalTest {
             options.body = new Buffer(0);
         }
 
-        queryString = options.url.split("?");
-        path = queryString.shift();
-        queryString = queryString.join("?");
+        path = options.url;
+        queryString = querystring.stringify(options.query);
+
+        if (options.parameters) {
+            Object.keys(options.parameters).forEach((key) => {
```

Comment:
This is just to fix the path for signing the request. I am just changing something like this `/blah/blah/token{?public}` to `/blah/blah/token`. So there is some overlap but it was the easiest way to fix the path for signing the request.